release/e2e.py

```diff
@@ -1490,6 +1490,7 @@ def run_test_config(
         "TEST_OUTPUT_JSON": results_json,
         "TEST_STATE_JSON": state_json,
         "IS_SMOKE_TEST": "1" if smoke_test else "0",
+        "RAY_lineage_pinning_enabled": "1",
```

Comment:
Can we split this part to a separate field and use update? 

```
    custom_env_vars = {
         "RAY_lineage_pinning_enabled": "1",
    }
    env_vars = {
        "RAY_ADDRESS": os.environ.get("RAY_ADDRESS", "auto"),
        "TEST_OUTPUT_JSON": results_json,
        "TEST_STATE_JSON": state_json,
        "IS_SMOKE_TEST": "1" if smoke_test else "0",
    }

    env_vars.update(custom_env_vars)
```

So that we can organize this better.